### DISCUSSION TITLE
Solidity 101, Fund me: are Github direct imports decentralised?

### DISCUSSION THREAD

**Discussion by encrypto95** [Cyfrin/foundry-full-course-cu#2754] [General]
Hi all!

I'm sorry if this has already been asked, but I can't find it anywhere.

My question is exactly what it says in the title: are github imports decentralized (explained in Lesson 10 of fund me)? It seems as - and I might be getting this wrong - if our beloved github platform fails there is no contract import possible. This seems like one of the risks associated with centralized applications.

Is the above really the case? If so, would that break smart contracts on the blockchain that import from github? Is there a security risk (associated to being decentralized) with using github imports?
I would almost think it can't be the case there is the single point failure risk, especially since we take the oracle problem so seriously.

I'm looking forward to your answer!

encrypto

**Comment by EngrPips** [comment]
Hello @encrypto95, The worst-case scenario will be that you request code from Github and you can't get it for your development, but any code you request from Github and use in your codebase and then deployed to the blockchain is stored on the blockchain, and your contract doesn't make any call to Github whenever you call a function on it because your contract actually store the code for its implementation and it doesn't fetch it from outside. So there is no Github risk associated with an already deployed contract, as far as I know, but yeah, there might be an issue where you can be unable to fetch code from Github for development, and that can be addressed by having a decentralized Github. Maybe you can work on that as a project.
